staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -349,12 +349,12 @@ func (c *Controller) handleObject(obj interface{}) {
 	if object, ok = obj.(metav1.Object); !ok {
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("error decoding object, invalid type"))
+			utilruntime.HandleErrorWithContext(context.Background(), nil, "Error decoding object, invalid type", "type", fmt.Sprintf("%T", obj), "value", obj)
```

Comment:
is it best practice to put a whole object into a (logging) key-value pair?

Reply:
I don't know about "best". It has been _my_ practice, on the grounds that more information is better for someone who has to debug the problem. Are you saying that there is a problem with this?

Reply:
The force-push to af7fb3cf4a3 adds a comment telling the developer to think about information disclosure risk here.

Reply:
This creates giant log lines potentially. Am not a fan.

@pohly are there any guidelines?

Reply:
I share the concern about information disclosure and log entry size. I'm not sure whether we have any such guidelines written down, though.

If it was some known type with known harmless fields, then dumping it may make sense. But for an unknown type I would not include the content of the object.

Reply:
The developer generally knows the type of object, so can usually make the call here.

The force-push to fdbf0bbb2d8 changes the code to omit the value and changes the comment to suggest adding the value based on size and sensitivity of content.